lib/ansible/playbook/base.py

```diff
@@ -343,7 +343,7 @@ def get_validated_value(self, name, attribute, value, templar):
         elif attribute.isa == 'float':
             value = float(value)
         elif attribute.isa == 'bool':
-            value = boolean(value, strict=False)
+            value = boolean(value, strict=True)
```

Comment:
It seems like this could potentially make a lot of things fail. Would it be better as a post validation method for no_log? Maybe safer, at least.

Reply:
For no_log, it seems clear that we should err on the side of caution. I'm just iffy over whether everything should become fatal immediately. Maybe a porting guide just to make note of the change?

Reply:
i ran with ci_complete and it actually found a bug in xml test, this is not full solution to no_log, just seems we are currently accepting 'bad false' for random strings in boolean fields and this should fix that.

Reply:
I was thinking more about playbooks doing odd things than our tests. But +1 from me. Could maybe be worth a note in the porting guide.

Reply:
updated

Reply:
updated